keps/sig-storage/284-enable-volume-expansion/README.md

```diff
@@ -344,3 +355,229 @@ type StorageClass struct {
+
+Enabling this feature should not negatively impact volume mount timings in general cases and hence percentile determined by `storage_operation_duration_seconds{operation_name=volume_mount}` metric should not change.
+
+Having said that if file system requires expansion during mount then it is obviously going to take longer for mount operation to finish.
+
+###### What are the SLIs (Service Level Indicators) an operator can use to determine the health of the service?
+
+- [ ] Metrics
+  - controller expansion operation duration:
+    - Metric name: storage_operation_duration_seconds{operation_name=expand_volume}
+    - [Optional] Aggregation method: percentile
+    - Components exposing the metric: kube-controller-manager
+  - controller expansion operation errors:
+    - Metric name: storage_operation_errors_total{operation_name=expand_volume}
+    - [Optional] Aggregation method: cumulative counter
+    - Components exposing the metric: kube-controller-manager
+  - node expansion operation duration:
+    - Metric name: storage_operation_duration_seconds{operation_name=volume_fs_resize}
+    - [Optional] Aggregation method: percentile
+    - Components exposing the metric: kubelet
+  - node expansion operation errors:
+    - Metric name: storage_operation_errors_total{operation_name=volume_fs_resize}
+    - [Optional] Aggregation method: cumulative counter
+    - Components exposing the metric: kubelet
+  - CSI operation metrics:
+    - Metric name: csi_sidecar_operations_seconds
+    - [Optional] Aggregation method: percentile
+    - Components exposing the metric: external-resizer
+
+- [ ] Other (treat as last resort)
+  - Details:
+
+###### Are there any missing metrics that would be useful to have to improve observability of this feature?
+
+### Dependencies
+
+<!--
+This section must be completed when targeting beta to a release.
+-->
+
+###### Does this feature depend on any specific services running in the cluster?
+
+This feature requires external-resizer running in the cluster for CSI volume expansion.
+
+### Scalability
+
+###### Will enabling / using this feature result in any new API calls?
+
+Yes enabling this feature requires new API calls.
+
+- Updates to PVs
+  - API operations
+    - PATCH PV
+    - GET PV
+    - List PVs
+  - originating components: kubelet, kube-controller-manager, external-resizer
+  - resync duration: 10mins
+- Update to PVCs:
+  - API operations
+    - PATCH PVC
+    - GET PVC
+    - List PVC
+  - originating components: kubelet, kube-controller-manager, external-resizer
+  - resync duration: 10mins
+
+###### Will enabling / using this feature result in introducing new API types?
+
+No
+
+###### Will enabling / using this feature result in any new calls to the cloud provider?
+
+Yes, we expect new calls to modify existing volume objects.
+
+###### Will enabling / using this feature result in increasing size or count of the existing API objects?
+
+Describe them, providing:
+  - API type(s): PVC
```

Comment:
We also have a `AllowVolumeExpansion` field in StorageClass.

Reply:
added